### ISSUE TITLE
Ineffective use of ensure_resource for packages

### ISSUE THREAD

**Issue by Wilm0r** [fraenki/puppet-galera#158]
Although `ensure_resource` is used to install `$additional_packages_real` at https://github.com/fraenki/puppet-galera/blob/master/manifests/init.pp#L502 , the added `before` clause still makes it impossible to avoid conflicts since duplicates should be exact matches.

In my setup, both the galera module and my main manifest try to install `rsync` (which I simply want on *every* machine in my fleet).

(My apologies if I'm just showing my ignorance on Puppet here. No clue whether this can be fixed by swapping the dependency to a `require` on the other end. On my side I can just remove the `rsync` install from my copy of this module.)

**Comment by fraenki** [comment]
> my main manifest try to install rsync (which I simply want on every machine in my fleet)

That's exactly my approach too, and I have to say that it works as expected in my case. Note that I use `ensure_packages()` to install rsync in my private manifest, so this might make a difference.

Of course, you could also set `manage_additional_packages` to `false` ([link](https://github.com/fraenki/puppet-galera/blob/master/REFERENCE.md#manage_additional_packages)). However, in this case you would need to ensure that you install all required packages for Galera/SST/etc.

**Comment by Wilm0r** [comment]
> That's exactly my approach too, and I have to say that it works as expected in my case. Note that I use ensure_packages() to install rsync in my private manifest, so this might make a difference.

That's intriguing..

```
        ensure_resource(
                "package",
                ["nano", "vim-tiny", "command-not-found"],
                {
                        ensure => purged,
                }
        )
        ensure_packages(
                ["nano", "vim-tiny", "command-not-found"],
                {
                        ensure => purged,
                        require => Package["joe"],
                }
        )
```

is accepted, so using `ensure_packages` as you mentioned. But replacing the second block with `ensure_resource`:

```
        ensure_resource(
                "package",
                ["nano", "vim-tiny", "command-not-found"],
                {
                        ensure => purged,
                        require => Package["joe"],
                }
        )
```

```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Function Call, Duplicate declaration: Package[nano] is already declared at (file: /etc/puppet/environments/test/manifests/site.pp, line: 244); cannot redeclare (file: /etc/puppet/environments/test/manifests/site.pp, line: 251) (file: /etc/puppet/environments/test/manifests/site.pp, line: 251, column: 2) on node tosca.beilen.gaast.net
```

> Of course, you could also set manage_additional_packages to false (link). However, in this case you would need to ensure that you install all required packages for Galera/SST/etc.

Yup, that was the first option I found but having to install the SQL/Galera-specific pkgs by hand misses the point of using a module especially also wrt versioning of course. :)

**Comment by fraenki** [comment]
@Wilm0r Following your observations... replacing `ensure_resources()` with `ensure_packages()` should fix the issue, right? I'd prepare a fix if you can confirm that this solves the issue.

**Comment by Wilm0r** [comment]
Tested one more thing:

```
        ensure_packages(
                ["nano", "vim-tiny", "command-not-found"],
                {
                        ensure => purged,
                }
        )
        ensure_packages(
                ["nano", "vim-tiny", "command-not-found"],
                {
                        ensure => purged,
                        require => Package["joe"],
                }
        )
```

Just in case ... but indeed, *two* uses of `ensure_packages` (instead of a mix of the two) work as well. Great. :)

**Comment by fraenki** [comment]
Released as 2.1.0.
